tensorflow/compiler/xla/service/gpu/tests/gpu_unrolling_test.cc

```diff
@@ -70,16 +70,26 @@ TEST_F(GpuUnrollingTest, UnrollFourTimes) {
   config.set_debug_options(debug_options);
   auto hlo_module = ParseAndReturnVerifiedModule(kAddModule, config).value();
 
-  CompileAndVerifyIr(std::move(hlo_module),
-                     R"(
+  const std::string pattern1 = R"(
```

Comment:
Is the issue that the vectorization occurs only with this hardware? Shouldn't we be matching unoptimized LLVM in this case?

Reply:
Fixed it for the UnrollFourTimes subtest but that did not work for the other subtests.

Reply:
For those other tests, the following IR line is failing:

; CHECK: load <4 x float>

Reply:
I changed the test a bit to make the matching of unoptimized LLVM work.  PTAL.

Reply:
I also discovered that the DisabledUnrollUnfusedSine and DisabledUnrollUnfusedCosine no longer call ROCm-Device-Lib functions but now generate LLVM so I updated those tests to remove the ROCm specific IR.

Reply:
Sorry, I pushed one more change to fix a typo and alter one of the subtests.